### PR TITLE
Add genesis hash in INTR message

### LINKED ISSUE #2325: Add genesis hash to intro packet

### PR DESCRIPTION
**Pull request by iketheadore**
Fixes #2325 

Changes:
- Add genesis hash in `INTR` message
- Update unit tests

Does this change need to mentioned in CHANGELOG.md?
Yes

### CODE REVIEW

**Comment by iketheadore** [comment]
OK, I will have a check

**Comment by iketheadore** [comment]
Do we need to add a `GenesisHash` field for `/api/v1/network/connections`?

**Comment by gz-c** [comment]
@iketheadore we don't need to, but I think we should, for full diagnostics

File: src/daemon/messages.go [hunk 1]
```diff
@@ -563,6 +571,15 @@ func (intro *IntroductionMessage) Verify(dc DaemonConfig, logFields logrus.Field
 		logger.WithError(err).WithFields(logFields).WithField("userAgent", userAgent).Warning("User agent is invalid")
 		return ErrDisconnectInvalidUserAgent
 	}
+	i += int(userAgentLen)
+
+	// var genesisHash cipher.SHA256
```
Comment (gz-c) [inline src/daemon/messages.go]: dead code

**Pull request merged**